### PR TITLE
feature 'apostrophe_as_package_separator'

### LINKED ISSUE #22644: Feature guard the removal of apostrophe as a package separator
0c81a5c9ca0287c0ee954a71b1bffdeb2cb3e9c8 merged the removal of `'` as a package separator.

After more discussion, the PSC believes it would be better to make the removal guarded by a feature.
(Similar to what was done with the `indirect`, `multidimensional` and `bareword_filehandles` features.)

That feature would be included in the `:default` feature bundle and all bundles up to `:5.40`. It wouldn't be included in `:5.42`. The main benefit would be that none of the unmaintained modules on CPAN and elsewhere would break, while newer code would not be able to the apostrophe any more.

It has been suggested that the only really surprising/problematic aspect of `'` as a package separator is in string interpolation (e.g. `"This is $owner's house"` would reference the package variable `$owner::s`). We think that if the feature is disabled, it might has well be disabled entirely, including for subroutines (e.g. `isn't` as provided by `Test::More` —Note: this specific subroutine has been deprecated in recent versions of [Test::More](https://metacpan.org/pod/Test::More#isnt)).

The proposed name for the feature (which is unlikely to ever be enabled manually, so there's no need to consider ease of typing): `apostrophe_as_package_separator`.

See also: [Putting apostrophe as a package separator behind a feature](https://www.nntp.perl.org/group/perl.perl5.porters/2024/10/msg268843.html).

### PR DESCRIPTION
**Pull request by tonycoz**
Revert removal of apostrophe in package names, and the deprecation of such.

Put the use of apostrophe in package names in source behind a default enabled feature.

This includes possible changes to the CPAN upstream modules to account for the brief set of development versions where apostrophe in names was not supported, but I could see an author ignoring it.  These version numbers may need to change depending on if/when this is merged.

Fixes #22644

---------------------------------------------------------------------------------
* This set of changes requires a perldelta entry, and it is included.
* This set of changes requires a perldelta entry, and I need help writing it.
* This set of changes does not require a perldelta entry.

### CODE REVIEW

**Review by jkeenan — APPROVED** [review]
I have not examined the actual code of this pull-request reversion.  I have attempted to examine the impact of this reversion on the set of CPAN distributions I typically look at for BBC breakage.

I pulled down this p.r. to disk (Ubuntu Linux 22.04 LTS), rebased it on blead, built a typical unthreaded `perl`, installed `cpanm` against that, then used `cpanm` to try to install various distributions from CPAN, including:

* Distributions whose PASS/FAIL status regressed between our July and August development releases (and their prerequisites)

* My "augmented CPAN-River-500" (500 distros)

These totalled approximately 661 distributions.  I found no adverse effects from this p.r.

* Distributions that originally FAILed due to apostrophe-in-names-removal and were left untouched since that went into effect and were not subsequently modified (*e.g.,* DBIx::Class) once again PASS.

* Distributions that originally FAILed due to apostrophe-in-names-removal but were subsequently updated by their authors/maintainers (*e.g.,* Try::Tiny; Email-Send) still PASS.

Subject to the limitations of the QA procedure described above, this pull request LGTM.

File: pod/perlmod.pod [hunk 1]
```diff
@@ -69,6 +69,21 @@ colon: C<$Package::Variable>.  If the package name is null, the
...
+single quote as separator, which was there to make Ada programmers feel
+like they knew what was going on.  Because the old-fashioned syntax is still
+supported for backwards compatibility, if you try to use a string like
+C<"This is $owner's house">, you'll be accessing C<$owner::s>; that is,
+the $s variable in package C<owner>, which is probably not what you meant.
+Use braces to disambiguate, as in C<"This is ${owner}'s house">.
+X<::> X<'>
+
+Using C<'> as a package separator is deprecated and will be removed in
+Perl 5.40.
```
Comment (ilmari) [inline pod/perlmod.pod]: This needs updating to reflect the new situation.

Comment (tonycoz) [reply pod/perlmod.pod]: I've changed it, but wasn't sure how much to mention here.

**Review by book — APPROVED** [review]

**Review by ap — APPROVED** [review]

**Comment by tonycoz** [comment]
Applied manually

**Pull request closed without merging**